### PR TITLE
Silence deprecated class warning

### PR DESCRIPTION
**Pull request by TsankoBg**
Replace `Requests` with `cURL`

### CODE REVIEW

**Comment by apeschar** [comment]
The class isn't deprecated, it's only been renamed to `WpOrg\Requests\Requests`.

See: https://github.com/WordPress/WordPress/blob/master/wp-includes/class-requests.php

Three options:

1. Define `REQUESTS_SILENCE_PSR0_DEPRECATIONS` to disable the warning until we can apply option (3).
2. Use `WpOrg\Requests\Requests` if it is available (using `class_exists` or by checking the WordPress version).
3. Always use `WpOrg\Requests\Requests` and drop support for WordPress versions below 6.2.0.

Simplest option (what I'd do) is (1) and requires adding this before the `Requests` class is first used:

~~~php
if (!defined('REQUESTS_SILENCE_PSR0_DEPRECATIONS')) {
  define('REQUESTS_SILENCE_PSR0_DEPRECATIONS', true);
}
~~~

Using cURL in a WordPress plugin is not ideal since it adds another requirement. The Requests class works without cURL, so [WordPress does not require or recommend it][wp].

[wp]: https://wordpress.org/about/requirements/

**Comment by TsankoBg** [comment]
I changed it based on your suggestion

File: common/src/API.php [hunk 1]
```diff
@@ -2,6 +2,10 @@
 namespace Valued\WordPress;
 
 use Exception;
+
+if (!defined('REQUESTS_SILENCE_PSR0_DEPRECATIONS')) {
+    define('REQUESTS_SILENCE_PSR0_DEPRECATIONS', true);
+}
 use Requests;
```
Comment (apeschar) [inline common/src/API.php]: `use` statements don't actually load anything, they just import a certain namespace.  So you can keep the `use` statements at the top as usual.

```suggestion
use Requests;

if (!defined('REQUESTS_SILENCE_PSR0_DEPRECATIONS')) {
    define('REQUESTS_SILENCE_PSR0_DEPRECATIONS', true);
}
```

**Review by apeschar — CHANGES_REQUESTED** [review]

**Comment by apeschar** [comment]
Released as 3.29. (But the [plugin page][wp] still shows the old version. I guess we need to wait for their cache to clear.)

[wp]: https://wordpress.org/plugins/trustprofile/

**Pull request merged**